src/modules/hip/kernel/tensor_stddev.hpp

```diff
@@ -0,0 +1,894 @@
+    int xDiff = xBufferLength - xAlignedLength;                                  // difference between bufferLength and alignedLength
+    uint srcIdx = ((id_z * xBufferLength) + id_x) * 3;
+
+    d_float24 src_f24;
+    rpp_hip_load24_pkd3_and_unpack_to_float24_pln3(srcPtr + srcIdx, &src_f24);   // load 24 pixels to local memory
+    if (id_x + 8 > xBufferLength)                                                // local memory reset of invalid values (from the vectorized global load) to 0.0f
+    {
+        for(int i = xDiff; i < 8; i++)
+        {
+            src_f24.f8[0].f1[i] = 0.0f;
+            src_f24.f8[1].f1[i] = 0.0f;
+            src_f24.f8[2].f1[i] = 0.0f;
+        }
+    }
+    src_f24.f8[0].f4[0] += src_f24.f8[0].f4[1];                                  // perform small work of vectorized float4 addition
+    src_f24.f8[1].f4[0] += src_f24.f8[1].f4[1];
+    src_f24.f8[2].f4[0] += src_f24.f8[2].f4[1];
+    partialRVar_smem[hipThreadIdx_x] = (src_f24.f8[0].f1[0] +
+                                        src_f24.f8[0].f1[1] +
+                                        src_f24.f8[0].f1[2] +
+                                        src_f24.f8[0].f1[3]);                    // perform small work of reducing R float4s to float using 1024 threads and store in _smem
+    partialGVar_smem[hipThreadIdx_x] = (src_f24.f8[1].f1[0] +
+                                        src_f24.f8[1].f1[1] +
+                                        src_f24.f8[1].f1[2] +
+                                        src_f24.f8[1].f1[3]);                    // perform small work of reducing G float4s to float using 1024 threads and store in _smem
+    partialBVar_smem[hipThreadIdx_x] = (src_f24.f8[2].f1[0] +
+                                        src_f24.f8[2].f1[1] +
+                                        src_f24.f8[2].f1[2] +
+                                        src_f24.f8[2].f1[3]);                    // perform small work of reducing B float4s to float using 1024 threads and store in _smem
+
+    __syncthreads();                                                             // syncthreads after _smem load
+
+    // Reduction of 1024 floats on 1024 threads per block in x dimension
+    for (int threadMax = 512; threadMax >= 1; threadMax /= 2)
+    {
+        if (hipThreadIdx_x < threadMax)
+        {
+            partialRVar_smem[hipThreadIdx_x] += partialRVar_smem[hipThreadIdx_x + threadMax];
+            partialGVar_smem[hipThreadIdx_x] += partialGVar_smem[hipThreadIdx_x + threadMax];
+            partialBVar_smem[hipThreadIdx_x] += partialBVar_smem[hipThreadIdx_x + threadMax];
+        }
+        __syncthreads();
+    }
+
+    // Final store to dst
+    if (hipThreadIdx_x == 0)
+    {
+        int totalElements = roiTensorPtrSrc[id_z].xywhROI.roiHeight * roiTensorPtrSrc[id_z].xywhROI.roiWidth;
+        float var = partialRVar_smem[0] + partialGVar_smem[0] + partialBVar_smem[0];
+        int idx = id_z * 4;
```

Comment:
This can be removed too if dstPtr can be received as a float4 *

Reply:
float4 is usually used to store float values but in this case we will need pointers which isn't feasible, is what I feel.

Reply:
@snehaa8 Not sure I understand, why can't we receive a float4*?

Reply:
please remove the template for stddev in outer api and for hip executor and change to float
and then inside use float4*

Reply:
Build is fine, but leads to hanging issue for unit and perf tests, so reverted change(wherein float* is converted to float4*)